dask/bag/core.py

```diff
@@ -1311,7 +1303,15 @@ def from_imperative(values):
 
 
 def merge_frequencies(seqs):
-    return list(merge_with(sum, map(dict, seqs)).items())
+    first, rest = seqs[0], seqs[1:]
+    if not rest:
+        return first
+    out = defaultdict(int)
+    out.update(first)
```

Comment:
Ah, I was incorrectly thinking of `merge_frequencies` as only computing at the very end.  I forgot that we were doing tree reductions.